File: apps/frontend/src/features/planning/types/milestone.ts
export type MilestoneStatus = "pending" | "in_progress" | "completed" | "skipped";

export interface Milestone {
  id: string;
  roadmap_id: string;
  org_id: string;
  title: string;
  description?: string;
  status: MilestoneStatus;
  order: number;
  created_at: string;
  updated_at: string;
}

File: apps/backend/src/modules/planning/domain/entities/milestone.py
from dataclasses import dataclass
from uuid import UUID

from src.modules.shared.domain.entity_v1 import Entity, TimestampedMixin
from src.modules.planning.domain.value_objects.milestone_status import MilestoneStatus


@dataclass(eq=False)
class Milestone(Entity, TimestampedMixin):
    roadmap_id: UUID
    org_id: UUID
    title: str
    description: str | None = None
    status: MilestoneStatus = MilestoneStatus.pending
    order: int = 0
    creator_id: UUID | None = None

    def change_status(self, new_status: MilestoneStatus) -> None:
        self.status = new_status
        self.touch()
